Doc/library/inspect.rst

```diff
@@ -348,26 +348,30 @@ attributes:
 
 .. function:: iscoroutinefunction(object)
 
-   Return ``True`` if the object is a :term:`coroutine function`
-   (a function defined with an :keyword:`async def` syntax).
+   Return ``True`` if the object is a :term:`coroutine function` (a function
+   defined with an :keyword:`async def` syntax), a :func:`functools.partial`
+   wrapping a :term:`coroutine function`, or an instance of a class defining
+   an :keyword:`async def` ``__call__``.
 
    .. versionadded:: 3.5
 
    .. versionchanged:: 3.8
       Functions wrapped in :func:`functools.partial` now return ``True`` if the
       wrapped function is a :term:`coroutine function`.
 
+   .. versionchanged:: 3.12
+      Instances of classes defining an :keyword:`async def` ``__call__`` now
+      return ``True``.
+
 
 .. function:: markcoroutinefunction(func)
 
    Decorator to mark a callable as a :term:`coroutine function` if it would not
    otherwise be detected by :func:`iscoroutinefunction`.
 
-   This may be of use for sync functions that return an awaitable or objects
-   implementing an :keyword:`async def` ``__call__``.
-
-   Prefer :keyword:`async def` functions or calling the function and testing
-   the return with :func:`isawaitable` where feasible.
+   This may be of use for sync functions that return a :term:`coroutine`, but
+   prefer :keyword:`async def` functions, or if necessary calling the function
+   and testing the return with :func:`iscoroutine` where feasible.
```

Comment:
This is a bit confusing because of the comma. Suggestion:

```suggestion
   This may be of use for sync functions that return a :term:`coroutine`,
   if the function is passed to an API that requires :func:`iscoroutinefunction`.

   When possible, using an :keyword:`async def` function is preferred.
   Also acceptable is calling the function
   and testing the return with :func:`iscoroutine`.
```

Feel free to reword this to your pleasing.

Reply:
Thanks @gvanrossum. 

I went with this. One option I looked at was: 

```
   When possible, using an :keyword:`async def` function is preferred.
   Alternatively, consider calling the function and testing the return with
   :func:`iscoroutine`, if feasible. 
```

The point was to capture your initial concerns from the discussions on the issue, and those linked from there. I took that to be ≈_"Do these other things if you can"_. I'm happy to defer to you on whether that's valuable (I guess it is!) and exact wording.